src/main/kotlin/org/amshove/kluent/Basic.kt

```diff
@@ -2,41 +2,44 @@ package org.amshove.kluent
 
 import org.junit.Assert.*
 import kotlin.reflect.KClass
+import kotlin.reflect.full.isSubclassOf
 
-infix fun Any?.shouldEqual(theOther: Any?) = assertEquals(theOther, this)
+infix fun <T> T.shouldEqual(theOther: T?): T = this.apply { assertEquals(theOther, this) }
```

Comment:
I really think we should start calling the parameter `expected` instead of `theOther`.